test/kt/esimd_radix_sort_by_key_out_of_place.cpp

```diff
@@ -0,0 +1,161 @@
+// -*- C++ -*-
+//===-- esimd_radix_sort_by_key_out_of_place.cpp --------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "../support/test_config.h"
+
+#include <oneapi/dpl/experimental/kernel_templates>
+#include <oneapi/dpl/iterator>
+
+#include <vector>
+#include <string>
+#include <cstdlib>
+
+#if __has_include(<sycl/sycl.hpp>)
+#    include <sycl/sycl.hpp>
+#else
+#    include <CL/sycl.hpp>
+#endif
+
+#include "../support/utils.h"
+#include "../support/sycl_alloc_utils.h"
+
+#include "esimd_radix_sort_utils.h"
+
+template <typename KeyT, typename ValueT, bool isAscending, std::uint32_t RadixBits, typename KernelParam>
+void
+test_sycl_buffer(sycl::queue q, std::size_t size, KernelParam param)
+{
+    std::vector<KeyT> input_keys(size);
+    std::vector<ValueT> input_values(size);
+    generate_data(input_keys.data(), size, 6);
+    generate_data(input_values.data(), size, 7);
+
+    std::vector<KeyT> expected_keys(input_keys);
+    std::vector<ValueT> expected_values(input_values);
+
+    std::vector<KeyT> actual_keys(expected_keys);
+    std::vector<ValueT> actual_values(expected_values);
+    std::vector<KeyT> actual_keys_out(size, KeyT{9});
+    std::vector<ValueT> actual_values_out(size, ValueT{9});
+
+    {
+        sycl::buffer<KeyT> keys(actual_keys.data(), actual_keys.size());
+        sycl::buffer<ValueT> values(actual_values.data(), actual_values.size());
+        sycl::buffer<KeyT> keys_out(actual_keys_out.data(), actual_keys_out.size());
+        sycl::buffer<ValueT> values_out(actual_values_out.data(), actual_values_out.size());
+
+        oneapi::dpl::experimental::kt::esimd::radix_sort_by_key_out_of_place<isAscending, RadixBits>(
+            q, keys, values, keys_out, values_out, param)
+            .wait();
+    }
+
+    auto expected_first = oneapi::dpl::make_zip_iterator(std::begin(expected_keys), std::begin(expected_values));
+    std::stable_sort(expected_first, expected_first + size, CompareKey<isAscending>{});
+
+    std::string parameters_msg = ", n: " + std::to_string(size) + ", sizeof(key): " + std::to_string(sizeof(KeyT)) +
```

Comment:
If you use `std::strstream` you may to build this message without a lot of `std::to_string` calls.

Reply:
fixed, thanks.